### ISSUE TITLE
Support detach method

### ISSUE THREAD

**Issue by ChanderG** [torch-spyre/torch-spyre#1084]
### What feature would you like to be added?

```
import torch
a = torch.zeros((20,), device="spyre", dtype=torch.float16)
a.detach()
```
leads to Segmentation fault:
```
/opt/rh/gcc-toolset-14/root/usr/include/c++/14/bits/stl_vector.h:1130: std::vector<_Tp, _Alloc>::reference std::vec
tor<_Tp, _Alloc>::operator[](size_type) [with _Tp = long int; _Alloc = std::allocator<long int>; reference = long i
nt&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
Signal Received: 6 (Aborted)
...
/dev/shm/workdir/.venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(+0x11fd21a0)[0x7fdf4f02e1a0]
/dev/shm/workdir/torch-spyre/torch_spyre/_C.so(+0x867d5)[0x7fde7c5387d5]
/dev/shm/workdir/torch-spyre/torch_spyre/_C.so(_ZN5spyre23get_device_stride_infosEN3c108ArrayRefIlEES2_NS_17SpyreTe
nsorLayoutESt6vectorIlSaIlEEb+0xbe5)[0x7fde7c53bf85]
/dev/shm/workdir/torch-spyre/torch_spyre/_C.so(_ZN5spyre12generate_dciB5cxx11EPKN2at6TensorENS_17SpyreTensorLayoutE
b+0x887)[0x7fde7c53cb57]
/dev/shm/workdir/torch-spyre/torch_spyre/_C.so(_ZN5spyre16create_dma_graphERKN2at6TensorES3_b+0x647)[0x7fde7c53da07
]
/dev/shm/workdir/torch-spyre/torch_spyre/_C.so(_ZN5spyre19copy_device_to_hostERKN2at6TensorES3_+0x3e)[0x7fde7c5432e
e]
...
```

### Why is this feature needed?

The in-tree-tests of test_reference_numerics test op values against a numpy reference. In this process, they use the detach() call on the target tensor (which is on spyre).
